OpenRA.Mods.Common/Pathfinder/PathGraph.cs

```diff
@@ -171,9 +173,14 @@ public List<GraphConnection> GetConnections(CPos position)
 
 		int GetCostToNode(CPos destNode, CVec direction)
 		{
-			var movementCost = locomotorInfo.MovementCostToEnterCell(worldMovementInfo, Actor, destNode, IgnoreActor, checkConditions);
-			if (movementCost != int.MaxValue && !(CustomBlock != null && CustomBlock(destNode)))
-				return CalculateCellCost(destNode, direction, movementCost);
+			var cost = locomotor.MovementCostForCell(destNode);
+
+			if (cost == int.MaxValue || !locomotorInfo.CanMoveFreelyInto(worldMovementInfo.World, Actor, destNode,
+				    IgnoreActor, checkConditions))
+				cost = int.MaxValue;
```

Comment:
Imho it would make more sense to just directly `return Constants.InvalidNode;` here, but not a blocker.